### ISSUE TITLE
uncertainties on Plot Any Two Expressions

### ISSUE THREAD

**Issue by adburnham** [CIRDLES/Squid#739]
Following up on #709 - though not in relation to the rejected points issue, but rather the way it calculates uncertainties on the slope and y-intercept. Different parts of SQUID seem to give different values (as outlined below), and also I thought you said in #709 that an update meant that Plot Any Two Expressions would give uncertainties.

Maybe I'm missing something, but although I can see the error envelope for the best fit of any two expressions (blue dotted line), the window only shows (written out) the values of the slope and y-intercept but not their uncertainties. Is there any way to find out the uncertainties, short of sending the values to Excel and calculating in Isoplot? I tried viewing it in "Manage expressions", where I can see the calculation of Expo, but the numbers in the Peek window don't match up to to what is calculated in "Plot Any Two Expressions".

**Comment by NicoleRayner** [comment]
I agree- you aren't missing anything the values don't show up (although it is inferred that this was coming in version 10.1 or 10.2 - I tried those just in case it appeared then disappeared but no dice).  As for the values being different, I believe this is related to slightly different calculations used within squid2.5 for the slope/Expo calculation than just RobReg.  I'd have to go through some of the documentation to confirm this though.  I'll have a quick look through and see if I can point you to a specific spot that explains this.

**Comment by bowring** [comment]
I don't see the problem.  Please provide an example.

**Comment by NicoleRayner** [comment]
v1.10.4, demo project:
![image](https://user-images.githubusercontent.com/40035242/204017306-82a86feb-918f-406d-9fbb-ac3d612c0f59.png)
value of slope and intercept given, but not uncertainties.  Are they somewhere else?

**Comment by bowring** [comment]
On the left window.

**Comment by NicoleRayner** [comment]
Those are the errors of the individual datapoints being regressed.  Looking for something like what is shown in the "expo" expression:
![image](https://user-images.githubusercontent.com/40035242/204020851-2621b1dc-cef6-409a-bfef-4b1019b29df2.png)
But in cases where the RobReg is using other X-Y parameters.  
I guess in theory you could make a custom expression with the stuff you are plotting in "Plot any two" using the RobReg and get the values from there @adburnham  although that doesn't address the discrepancies based on what regression algorithm is being used.  See issues #696 and #617 for some context

**Comment by bowring** [comment]
Well, I mis-understood issue #709 originally and addressed it by providing the point-wise uncertainties in the left hand details.  Now I understand what is wanted.

History:  The plot-any-two functionality uses an algorithm developed by @noahmclean and implemented at https://github.com/CIRDLES/McLeanRegression.  The corresponding paper is [here](https://github.com/CIRDLES/McLeanRegression/blob/master/src/main/resources/McLean%20-%20Straight%20line%20regression%20through%20data%20with%20correlated%20uncertainties%20in%20two%20or%20more%20dimensions%2C%20with%20an%20application%20to%20kinetic%20isotope%20fractionation%20(GCA%202014).pdf).
 
The uncertainty is represented by the envelope, which has a plus/minus value at each x.  The original matlab code is [here](https://github.com/CIRDLES/McLeanRegression/tree/master/src/main/resources/MatLabCode).

The outputs are described there as:
```
'a' is the point on the line with first component a1 specified above
'a2s' contains the two-sigma uncertainties in the free components of a
'v' is the direction vector of the line with first component v1 as input
'v2s' contains the two-sigma uncertainties in the free components of v
'Sav' contains the covariance matrix for the free components of a and v
'MSWD' is the mean of the squared weighted deviates (reduced chi-square)
'n' is the number of analyses included in the calculation, = sum(skipv)
```

Please review and decide how you want to proceed.  If you want a simple vanilla regression, then that should become an issue asking for that alternative.  Hopefully, @noahmclean will chime in.

Cheers

**Comment by adburnham** [comment]
This error arose as I was writing up the results section of a report. Because there is a range of opinions about the best fit slope (some researchers finding 1.7 better than the previously preferred value of 2.0) I thought it would be good to report what I found. I think it would be good to have consistency between the Expo value used for calculations and the slope calculated by Plot Any Two Expressions.
